source/common/upstream/health_checker_impl.cc

```diff
@@ -793,6 +793,9 @@ void GrpcHealthCheckerImpl::GrpcActiveHealthCheckSession::logHealthCheckStatus(
     case grpc::health::v1::HealthCheckResponse::UNKNOWN:
       service_status = "unknown";
       break;
+    case grpc::health::v1::HealthCheckResponse::SERVICE_UNKNOWN:
+      service_status = "service_unknown";
+      break;
```

Comment:
I don't now the details of proto parsing, but I'm pretty sure that a different enum value might still enter the default statement below and we need to remove that default and/or remove the crash? Can you add a test that forces a different unknown enum value using a static cast?

Reply:
Hum according to https://github.com/grpc/grpc/blob/master/src/proto/grpc/health/v1/health.proto#L37 there is no other enum. Only way it could be achieved would be by crafting a custom grpc request I guess. Not sure what the behaviour should be like on envoy side then

Reply:
See my comment below. I suspect that will crash also. If so can you fix that as well?

Reply:
That will crash for sure, but my point being I guess the crash was there for a reason 🤔 but I'm fine with adding a failsafe 😄 I'll fix it tomorrow!

Reply:
Yes if we don't have a higher layer PGV validation on this field, we have to protect against an upstream sending us an enum that we don't know about, so we definitely need to handle/test this case. Thank you!

/wait